### PR TITLE
fix(components): fixed various design issues across components

### PR DESCRIPTION
**Pull request by YonathanMeguira**
Fixed elements on this list:
https://docs.google.com/document/d/1QUQjbJJ6_02ZC3NWgsgI9bvZFYKOv7bQPrVjyGcfB8Q/edit

### CODE REVIEW

File: lib/src/button/button.component.scss [hunk 1]
```diff
@@ -62,4 +63,9 @@
 :host.dato-icon-button[datoSize='sm'] button {
   width: 16px;
   height: 16px;
-}
\ No newline at end of file
+}
+
+:host ::ng-deep dato-icon {
```
Comment (bh86) [inline lib/src/button/button.component.scss]: redundant

File: lib/src/button/button.component.ts [hunk 2]
```diff
@@ -33,6 +35,15 @@ export class DatoButtonComponent {
 
   ngOnInit() {
     const button = query('button', this.host.nativeElement);
+    const firstChild = button.childNodes[0] as HTMLElement;
```
Comment (bh86) [inline lib/src/button/button.component.ts]:
```
if ($0.childNodes.length > 1) {
	if ($0.childNodes[0].tagName === 'dato-icon') {
		$0.childNodes[0].classList.add('mr-4');
	}
	else if ($0.childNodes[$0.childNodes.length - 1].tagName === 'dato-icon') {
		$0.childNodes[$0.childNodes.length - 1].classList.add('ml-4');
	}
}
```

File: lib/src/dialog/content/dialog-content.component.scss [hunk 3]
```diff
@@ -2,6 +2,7 @@
   padding: 0 30px;
   padding-bottom: 30px;
   overflow: auto;
+  font-size: 14px;
```
Comment (bh86) [inline lib/src/dialog/content/dialog-content.component.scss]: Use the mixin:
```
@include typography(simple);
```

File: lib/src/input/input.component.scss [hunk 4]
```diff
@@ -27,17 +27,29 @@
 
 :host[datoSize='sm'] dato-icon {
   right: 3px;
+  height: 18px;
+}
+
+:host[datoSize='md'] dato-icon {
```
Comment (bh86) [inline lib/src/input/input.component.scss]: Merge  `md` and `lg` selectors, under the same rule.

File: lib/src/input/input.theme.scss [hunk 5]
```diff
@@ -36,4 +35,10 @@ dato-input:hover dato-icon path {
   100% {
     transform: rotate(360deg);
   }
-}
\ No newline at end of file
+}
+
+dato-input {
```
Comment (bh86) [inline lib/src/input/input.theme.scss]: That's should be already implemented. Look at `input-helper.scss`, in the `input-disabled` mixin.
If you need to fix the input, put it there.

File: lib/src/list/search.strategy.ts [hunk 6]
```diff
@@ -14,7 +14,7 @@ export interface DatoListSearchStrategy {
 }
 
 /**
- * The default search strategy is compare lowercased strings
+ * The default search stratrgy is compare lowercased strings
```
Comment (bh86) [inline lib/src/list/search.strategy.ts]: typo :)

File: lib/src/select/select.theme.scss [hunk 7]
```diff
@@ -1,30 +1,29 @@
...
+.dato-select__placeholder,
+.dato-select__option--disabled,
+dato-select__not-found,
+.dato-select__group {
   color: themify(primary-300);
 }
 
 .dato-select__group-container:not(:first-child) .dato-select__group {
-  border-top: themify(primary-200) 1px solid;
+  border-top: solid 1px themify(primary-200);
```
Comment (bh86) [inline lib/src/select/select.theme.scss]: Change it back so the `solid 1px` will be last. This is fixing an issue in Safari, see:
https://github.com/datorama/themify#known-issues

File: lib/src/tabs/tabs.component.scss [hunk 8]
```diff
@@ -55,4 +56,11 @@ $tab-header-y-padding: 5px;
   .tab-pointer {
     display: none;
   }
+}
+
+.tab-md {
```
Comment (bh86) [inline lib/src/tabs/tabs.component.scss]: Use the `typography` mixin.

File: lib/src/dialog/content/dialog-content.component.scss (hunk 3 above)
Comment (YonathanMeguira) [reply lib/src/dialog/content/dialog-content.component.scss]: nice !

File: lib/src/select/select.theme.scss (hunk 7 above)
Comment (YonathanMeguira) [reply lib/src/select/select.theme.scss]: done

File: lib/src/list/search.strategy.ts (hunk 6 above)
Comment (YonathanMeguira) [reply lib/src/list/search.strategy.ts]: done

File: lib/src/tabs/tabs.component.scss (hunk 8 above)
Comment (YonathanMeguira) [reply lib/src/tabs/tabs.component.scss]: done

File: lib/src/input/input.theme.scss (hunk 5 above)
Comment (YonathanMeguira) [reply lib/src/input/input.theme.scss]: done

File: lib/src/input/input.component.scss (hunk 4 above)
Comment (YonathanMeguira) [reply lib/src/input/input.component.scss]: done

File: lib/src/dialog/content/dialog-content.component.scss (hunk 3 above)
Comment (YonathanMeguira) [reply lib/src/dialog/content/dialog-content.component.scss]: done

File: lib/src/button/button.component.ts (hunk 2 above)
Comment (YonathanMeguira) [reply lib/src/button/button.component.ts]: done

File: lib/src/button/button.component.scss (hunk 1 above)
Comment (YonathanMeguira) [reply lib/src/button/button.component.scss]: done

**Pull request merged**